01/web-client/main.ts

```diff
@@ -101,41 +103,111 @@ function tryReadWord(lines: Line[]) {
 	}
 	// find the first line that is visible within the viewport
 	const readableLine = unfinishedLines.find(
-		line => (
-			line.borderBox.top > 0 &&
-			line.borderBox.bottom < window.innerHeight
```

Comment:
The problem of `line.borderBox.bottom < window.innerHeight` was that when the lines start exceeding the viewport height, this will never be true anymore. Scrolling wasn't supported.